src/gfx/gizmos/gizmo_input.cpp

```diff
@@ -17,20 +18,18 @@ void InputContext::begin(const InputState &inputState, ViewPtr view) {
   heldHandleUpdated = false;
 }
 
+// TODO: move raycasting out of updateHandle, maybe even call it with a distance parameter
+// let the gizmo itself handle the raycasting however it needs to instead perhaps
 void InputContext::updateHandle(Handle &handle) {
```

Comment:
Yeah I was thinking, instead of using inheritance and then calling `handle.resolveHover`, the InputContext can instead have a function `addRaycastHit(Handle&, float distance)` since the raycast happens during this call right now, it wouldn't be that much different to do the raycast outside of InputContext and just give it the results.

So usage in the translation gizmo will be something like this:
```cpp
  auto &box = ...;
  float4x4 invHandleTransform = linalg::inverse(...);
  float3 rayLoc1 = linalg::mul(invHandleTransform, float4(eyeLocation, 1)).xyz();
  float3 rayDir1 = linalg::mul(invHandleTransform, float4(rayDirection, 0)).xyz();
  float2 hit = intersectAABB(rayLoc1, rayDir1, box.min, box.max);
  if (hit.x < hit.y) {
   inputContext.addRaycastHit(handle, hit.x); // <- call here
  }
```
Handle is only used to identify and persist the held state of some control then.
Of course the above raycast can then be refactored into a function as well so other gizmos can use the same logic.

Reply:
Right, that makes sense, I'll work on refactoring to that instead, likely I'll shift the raycast logic into `gizmo_math.hpp` to be used across gizmos. Thanks for the pointers!